pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,871 @@
+    # isSingle==False means a model or block
+    isSingle = False
+
+    if isinstance(pyomoElement, pyo.Objective):
+        objectives = [pyomoElement]
+        constraints = []
+        expressions = []
+        templatize_fcn = templatize_constraint
+        useAlignEnvironment = False
+        isSingle = True
+
+    elif isinstance(pyomoElement, pyo.Constraint):
+        objectives = []
+        constraints = [pyomoElement]
+        expressions = []
+        templatize_fcn = templatize_constraint
+        useAlignEnvironment = False
+        isSingle = True
+
+    elif isinstance(pyomoElement, pyo.Expression):
+        objectives = []
+        constraints = []
+        expressions = [pyomoElement]
+        templatize_fcn = templatize_expression
+        useAlignEnvironment = False
+        isSingle = True
+
+    elif isinstance(pyomoElement, ExpressionBase):
+        objectives = []
+        constraints = []
+        expressions = [pyomoElement]
+        templatize_fcn = templatize_passthrough
+        useAlignEnvironment = False
+        isSingle = True
+
+    elif isinstance(pyomoElement, _BlockData):
+        objectives = [
+            obj
+            for obj in pyomoElement.component_data_objects(
+                pyo.Objective, descend_into=True, active=True
+            )
+        ]
+        constraints = [
+            con
+            for con in pyomoElement.component_objects(
+                pyo.Constraint, descend_into=True, active=True
+            )
+        ]
+        expressions = []
+        templatize_fcn = templatize_constraint
+
+    else:
+        raise ValueError(
+            "Invalid type %s passed into the latex printer" % (str(type(pyomoElement)))
+        )
+
+    # In the case where just a single expression is passed, add this to the constraint list for printing
+    constraints = constraints + expressions
+
+    # Declare a visitor/walker
+    visitor = _LatexVisitor()
+    visitor.disableSmartVariables = disableSmartVariables
+    # visitor.xOnlyMode = xOnlyMode
+
+    # # Only x modes
+    # # Mode 0 : dont use
+    # # Mode 1 : indexed variables become x_{_{ix}}
+    # # Mode 2 : uses standard alphabet [a,...,z,aa,...,az,...,aaa,...] with subscripts for indices, ex: abcd_{ix}
+    # # Mode 3 : unwrap everything into an x_{} list, WILL NOT WORK ON TEMPLATED CONSTRAINTS
+
+    nameReplacementDict = {}
+    if not isSingle:
+        # only works if you can get the variables from a block
+        variableList = [
+            vr
+            for vr in pyomoElement.component_objects(
+                pyo.Var, descend_into=True, active=True
+            )
+        ]
+        if xOnlyMode == 1:
+            newVariableList = ['x' for i in range(0, len(variableList))]
+            for i in range(0, len(variableList)):
+                newVariableList[i] += '_' + str(i + 1)
+            overwriteDict = dict(zip([v.name for v in variableList], newVariableList))
```

Comment:
Wondering if this dictionary should really be a `ComponentMap` and that the keys should by pyomo components (rather than their names). Also, do you think we could support naming Sets and Params out of it as well? Sets are the hard ones because there's what you call the set, and then what you use for indices. But perhaps to start we could have people specify a list of index names and if you run out you can just yell at them. Thinking of cases like:
```
@m.Constraint(m.TIME)
def something(m, t):
    return sum(m.some_var[t_prime] for t_prime in m.TIME if t_prime < t) == 1
```
You'd want `overwriteDict[m.TIME] = ('\mathcal T', ['t', "t'"])` or something like that...